modules/videopress/class.videopress-xmlrpc.php

```diff
@@ -155,18 +155,25 @@ public function update_poster_image( $request ) {
 			return false;
 		}
 
+		// Switch to photon if needed.
+		$poster = str_replace(
+			'https://videos.files.wordpress.com',
+			'https://i0.wp.com/videos.files.wordpress.com',
```

Comment:
Makes sense, I'll apply this change and test. I was really just trying to fix the fact that Calypso won't work without Photon URLs here. But it makes sense to use the internal methods for this.